halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/ProviderValidator.java

```diff
@@ -18,20 +18,33 @@
 
 package com.netflix.spinnaker.halyard.config.validate.v1.providers;
 
+import com.netflix.spinnaker.halyard.config.model.v1.node.Account;
 import com.netflix.spinnaker.halyard.config.model.v1.node.Provider;
 import com.netflix.spinnaker.halyard.config.model.v1.node.Validator;
 import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemSetBuilder;
 import com.netflix.spinnaker.halyard.core.problem.v1.Problem;
 import org.springframework.stereotype.Component;
 
+import java.util.List;
+import java.util.stream.Collectors;
+
 @Component
 public class ProviderValidator extends Validator<Provider> {
   @Override
+  @SuppressWarnings("unchecked")
   public void validate(ConfigProblemSetBuilder p, Provider n) {
     if (n.isEnabled()) {
       if (n.getAccounts() == null || n.getAccounts().isEmpty()) {
         p.addProblem(Problem.Severity.WARNING, "Provider " + n.getNodeName() + " is enabled, but no accounts have been configured.");
       }
+
+      List bootstrapAccounts = (List) n.getAccounts()
+                                       .stream()
+                                       .filter(a -> ((Account) a).getBootstrapOnly())
+                                       .collect(Collectors.toList());
+      if (bootstrapAccounts.size() >= 2) {
```

Comment:
You're only validating that there is at most 1 bootstrap-only account per provider. You want to do this check in the `Providers` class and validate across all accounts